platform/ios/src/UIColor+MGLAdditions.mm

```diff
@@ -25,13 +25,48 @@ + (UIColor *)mgl_colorWithColor:(mbgl::Color)color
 @implementation NSExpression (MGLColorAdditions)
 
 + (NSExpression *)mgl_expressionForRGBComponents:(NSArray<NSExpression *> *)components {
+    if (UIColor *color = [self mgl_colorWithRGBComponents:components]) {
+        return [NSExpression expressionForConstantValue:color];
+    }
+    
     NSExpression *color = [NSExpression expressionForConstantValue:[UIColor class]];
-    return [NSExpression expressionForFunction:color selectorName:@"colorWithRed:green:blue:" arguments:components];
+    NSExpression *alpha = [NSExpression expressionForConstantValue:@1.0];
+    return [NSExpression expressionForFunction:color
+                                  selectorName:@"colorWithRed:green:blue:alpha:"
+                                     arguments:[components arrayByAddingObject:alpha]];
 }
 
 + (NSExpression *)mgl_expressionForRGBAComponents:(NSArray<NSExpression *> *)components {
+    if (UIColor *color = [self mgl_colorWithRGBComponents:components]) {
+        return [NSExpression expressionForConstantValue:color];
+    }
+    
     NSExpression *color = [NSExpression expressionForConstantValue:[UIColor class]];
-    return [NSExpression expressionForFunction:color selectorName:@"colorWithRed:green:blue:alpha:" arguments:components];
+    return [NSExpression expressionForFunction:color
+                                  selectorName:@"colorWithRed:green:blue:alpha:"
+                                     arguments:components];
+}
+
++ (UIColor *)mgl_colorWithRGBComponents:(NSArray<NSExpression *> *)components {
+    if (components.count < 3 || components.count > 4) {
+        return nil;
+    }
+    
+    for (NSExpression *component in components) {
+        if (component.expressionType != NSConstantValueExpressionType) {
+            return nil;
+        }
```

Comment:
Is this requiring the components of an RGB(A) expression to be literals?  If so, note that that isn't required by the core `"rgb"` / `"rgba"` expressions.

Reply:
This method is a convenience. If it returns `nil`, then the caller will synthesize a `colorWithRed:green:blue:alpha:` function expression instead, which can take arbitrary expressions as the components.